rest/routing.go

```diff
@@ -293,7 +295,18 @@ func CreateAdminRouter(sc *ServerContext) *mux.Router {
 	// The routes below are part of the CouchDB REST API but should only be available to admins,
 	// so the handlers are moved to the admin port.
 	r.Handle("/{newdb:"+dbRegex+"}/",
-		makeHandler(sc, adminPrivs, []Permission{PermCreateDb}, nil, (*handler).handleCreateDB)).Methods("PUT")
+		makeHandlerSpecificAuthScope(sc, adminPrivs, []Permission{PermCreateDb}, nil, (*handler).handleCreateDB, func(bodyJSON []byte) (string, error) {
```

Comment:
I think I'd prefer to define the auth scope function elsewhere (in admin_api beside handleCreateDB, maybe?), to keep router.go more readable.